float2d new_goals
---
bool success
string message